classes/webservice/ApiNode.php

```diff
@@ -0,0 +1,268 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+class ApiNode
+{
+    public const TYPE_VALUE = 'value';
+    public const TYPE_LANGUAGE = 'language';
+    public const TYPE_PARENT = 'parent';
+    public const TYPE_LIST = 'list';
+
+    public static $languages;
+    private $type;
+    private $name;
+    private $value;
+    private $values;
+    private $attributes;
+    private $nodes;
+
+    private function __construct($type, $name = null, $value = null, $attributes = [], $nodes = [])
```

Comment:
```suggestion
    public function __construct($type, $name = null, $value = null, $attributes = [], $nodes = [])
```

Reply:
This is private on purpose. Objects are created through static functions ApiNode::parent or ApiNode::list (or ApiNode::value and ApiNode::lang, privately)

Reply:
OK. 

Could you strict type the constructor ?

Reply:
Done in b4d8238179f092e7c110675468ba11b74c4eebf2